code/game/objects/items/gem_items.dm

```diff
@@ -0,0 +1,1245 @@
+/obj/item/carvedgem
+	name = "the humble bug beaver"
+	desc = "Oh for the love of twigs, looks like something messed up! tell a dam coder about this! TOODLES!"
+	icon = 'icons/roguetown/gems/gem_amber.dmi'
+	icon_state = "beaver_amber"
+	grid_height = 1
+	grid_width = 1
+	sellprice = 0
+
+//.....................Shell......................//
+
+/obj/item/carvedgem/shell/openoyster
```

Comment:
Make a /obj/item/carvedgem/shell and put the icon on it instead of rewriting it everytime, SAME FOR ALL YOUR UNIQUE THINGS like /jade /rosa